src/model/mod.rs

```diff
@@ -429,33 +521,60 @@ pub fn place_ships(mut p1: &mut types::Player)
     let s4 = types::ShipType{ name: "Battleship".to_string(), size: 5, amount: 1 };
     let ships = vec![s1, s2, s3, s4];
 
-    // In a replacement situation for p2,
-    // p1 doesn't need to place the ships again.
-    if p1.capacity == 0 {
+    if p.capacity == 0 && p.name == "AI" {
 
-        print_boards(&p1);
+        // Holds the remaining indices to place a ship at.
+        let mut vec = Vec::new();
+        for i in 0..100 {
+            vec.push(i);
+        }
 
-        // Asks player1 to place the ships.
+        // Asks the AI to place its ships.
         for i in ships.iter() {
             for _ in 0..i.amount {
                 loop {
-                    println!("{}, please enter the first coordinate for your {:?} ({}{}",
-                        p1.name, i.name, i.size, " fields).");
-                    match place(&mut p1, i) {
+                    match place_ai(&mut p, i, &mut vec) {
                         Ok(_) => { break; },
                         Err(e) => {
                             match e {
-                                types::ErrorType::InvalidField => {
-                                    println!("Invalid position for this ship, {}",
-                                        "please choose another coordinate.");
-                                },
-                                types::ErrorType::DeadEnd => {},
+                                types::ErrorType::InvalidField => {},
+                                types::ErrorType::DeadEndAI => { return Err(e) },
+                                _ => { return Err(e) },
                             }
                         },
                     }
                 }
-                p1.capacity += i.size;
-                print_boards(&p1);
+                p.capacity += i.size;
+            }
+        }
+    } else {
+        if p.capacity == 0 {
+
+            print_boards(&p);
+
+            // Asks the human player to place the ships.
+            for i in ships.iter() {
+                for _ in 0..i.amount {
+                    loop {
+                        println!("{}, please enter the first coordinate for your {:?} ({}{}",
+                            p.name, i.name, i.size, " fields).");
+                        match place(&mut p, i) {
+                            Ok(_) => { break; },
+                            Err(e) => {
+                                match e {
+                                    types::ErrorType::InvalidField => {
+                                        println!("Invalid position for this ship, {}",
+                                            "please choose another coordinate.");
+                                    },
+                                    types::ErrorType::DeadEndHuman => { return Err(e) },
+                                    _ => { return Err(e) },
```

Comment:
You could delete the case `types::ErrorType::DeadEndHuman` as the right hand side is equal to the default-case.